### ISSUE TITLE
Add Tuple Functions

### ISSUE THREAD

**Issue by cranst0n** [spebbe/dartz#89]
I think it would be useful to add the following functions to the `TupleN` classes:

* head: Returns the first element of the tuple
* tail: Return all but the first elements
* init: Return all elements except the last element
* last: Returns the last element.

I've implemented these for few of them using extensions but can open a PR for all of them implemented directly in the class if you would be willing to merge.

Here's a sample of what I've done thus far:

```dart
extension Tuple3Ops<A, B, C> on Tuple3<A, B, C> {
  A get head => value1;
  Tuple2<B, C> get tail => Tuple2(value2, value3);

  Tuple2<A, B> get init => Tuple2(value1, value2);
  C get last => value3;
}
```

The `head`/`last` would just be different names for existing functionality, so if you think it's unnecessary I would leave them out. Any other feedback would be welcome.

**Comment by spebbe** [comment]
Hi @cranst0n – sounds like a great addition that would bring us one step closer to https://github.com/milessabin/shapeless `HList`s! Your naming scheme looks good to me. Please open a PR if you are willing. Thank you!

**Comment by cranst0n** [comment]
Great! I'll submit something in the next few days.

**Comment by spebbe** [comment]
Merged and released in 0.10.1 – Thanks again!
